client/client-lib/src/query.rs

```diff
@@ -147,30 +154,48 @@ impl<R: Hash + Eq + Clone> QueryStrategy<R> for EventuallyConsistent<R> {
 
 /// Returns when `required` responses are equal
 pub struct CurrentConsensus<R> {
-    pub results: HashMap<R, HashSet<PeerId>>,
+    /// Previously received responses/results
+    ///
+    /// Since we don't expect a lot of different responses,
+    /// it's easier to store them in `Vec` and do a linear search
+    /// than required `R: Ord` or `R: Hash`.
+    pub existing_results: Vec<(R, HashSet<PeerId>)>,
     pub errors: HashMap<PeerId, JsonRpcError>,
     required: usize,
 }
 
 impl<R> CurrentConsensus<R> {
     pub fn new(required: usize) -> Self {
         Self {
-            results: HashMap::new(),
+            existing_results: vec![],
             errors: HashMap::new(),
             required,
         }
     }
 }
 
-impl<R: Hash + Eq + Clone> QueryStrategy<R> for CurrentConsensus<R> {
+impl<R: Eq + Clone + Debug> QueryStrategy<R> for CurrentConsensus<R> {
     fn process(&mut self, response: FedResponse<R>) -> QueryStep<R> {
         match response {
             FedResponse {
                 peer,
                 result: Ok(result),
             } => {
-                let peers = self.results.entry(result).or_insert_with(HashSet::new);
-                peers.insert(peer);
+                let mut found = false;
+                for (prev_result, peers) in &mut self.existing_results {
```

Comment:
Same: Please review carefully, not sure if we have good tests for this.

Reply:
I think you could use `self.existing_results.iter_mut().find(…)` to make this a lot more readable

Reply:
Done.